Source/Wikitivity.Agent/WikitivityAgentHelper.cs

```diff
@@ -101,28 +214,41 @@ public List<WikitivityUploadsAgent.DataObtainedSingleRequestObject> PopulateArti
 
 		public static string GetInstanceUrl(IAgentHelper helper, AgentBase aB)
 		{
-			string instanceUrl = "";
 			try
 			{
 				aB.RaiseMessage("Attempting to get URL...", 1);
-				instanceUrl = helper.GetInstanceSettingBundle().GetString("Relativity.Core", "KeplerServicesUri");
+				//instanceUrl = helper.GetInstanceSettingBundle().GetString("Relativity.Core", "KeplerServicesUri");
+				string instanceUrl = helper.GetInstanceSettingBundle().GetStringAsync("Relativity.Wikitivity", "WikitivityWebApi").Result;
+
 				aB.RaiseMessage("Acquired URL: " + instanceUrl, 1);

```

Comment:
all raise messages are visible to the Relativity user using the app. I would use them carefully. 
For troubleshooting, i would use the logging frame to log stuff.